Update cicd docs (#502)
kyma-project/istio: docs/contributor/04-30-ci-cd.md

## Pipelines running on development pull requests

The following CI jobs are part of the development cycle.

| Name                                                                                                                                                                    | Description                                                                                                                                            |
|-------------------------------------------------------------------------------------------------------------------------------------------------------------------------|--------------------------------------------------------------------------------------------------------------------------------------------------------|
| [`Wait for image build`](https://github.com/kyma-project/istio/blob/main/.github/workflows/pull-integration.yaml#L22)                                                   | Waits for the operator's image build to finish so that the image-dependent jobs can be executed.                                                       |

> | [`pull-istio-operator-build`](https://github.com/kyma-project/test-infra/blob/main/prow/jobs/kyma-project/istio/istio-manager.yaml#L51)                                 | Builds Kyma Istio Operator's image and pushes it to the `dev` registry.                                                                                |

Author: right

# CI/CD
The Istio module contains several pipelines that run on pull requests, on the main branch, on the release branch, and periodically. They perform various tasks such as running a linter, executing unit tests with code coverage, generating artifacts for development and production purposes, and running integration tests. The integration tests ensure the functional correctness of Kyma Istio Operator and verify if upgrading to the new version of Kyma Istio Operator is possible.

> ## Pipelines running on development pull requests

Reviewer: We should ahve it similar to [api-gateway](https://github.com/kyma-project/api-gateway/blob/main/docs/contributor/04-30-ci-cd.md#pipelines-running-on-pull-requests-to-the-main-branch)

```suggestion
## Pipelines running on pull requests to the `main` branch
```

| [`main integration tests / Wait for image build`](https://github.com/kyma-project/istio/blob/26d8cfa69b91828fc34afd3b6aaf63c854d0cc54/.github/workflows/main-integration.yaml#L20)            | Waits for the operator's image build to finish so that image-dependent jobs can be executed.                                                      |
| [`main integration tests / Istio upgrade integration tests`](https://github.com/kyma-project/istio/blob/26d8cfa69b91828fc34afd3b6aaf63c854d0cc54/.github/workflows/main-integration.yaml#L30) | Executes the upgrade integration test suite of Istio Module and verifies if the existing release can be successfully upgraded to the new version. |
| [`main integration tests / Slack Notification`](https://github.com/kyma-project/istio/blob/26d8cfa69b91828fc34afd3b6aaf63c854d0cc54/.github/workflows/main-integration.yaml#L44)              | Sends a Slack notification to the team's channel if any pipelines fail on the main branch.                                                        |

> ## Pipelines running on the PR during the release process

Reviewer: We should have it similar to [api-gateway](https://github.com/kyma-project/api-gateway/blob/main/docs/contributor/04-30-ci-cd.md#pipelines-running-on-pull-requests-to-release-branches).
```suggestion
## Pipelines running on pull requests to release branches
```
